spec/calculator_spec.rb

```diff
@@ -1 +1,214 @@
+    end
+
+    it "adds two floats" do
+      expect(calculator.add(1.2, 2.6)).to be_within(0.5).of(4)
+    end
+
+    it "adds a float and an integer" do
+      expect(calculator.add(1.2, 2)).to be_within(0.5).of(3)
+    end
+  end
+
+  describe "#subtract" do
+    it "raises error if only one value is passed as argument" do
+      expect{ Calculator.new.subtract(1) }.to raise_error(ArgumentError)
+    end
+
+    it "raises error if more than two values are passed as argument" do
+      expect{ Calculator.new.subtract(1, 2, 3) }.to raise_error(ArgumentError)
+    end
+
+    it "subtracts two positive numbers" do
+      expect(calculator.subtract(3, 1)).to eq(2)
+    end
+
+    it "subtracts two negative numbers" do
+      expect(calculator.subtract(-1, -2)).to eq(1)
+    end
+
+    it "subtracts a positive number and a negative number" do
+      expect(calculator.subtract(1, -2)).to eq(3)
+    end
+
+    it "subtracts two floats" do
+      expect(calculator.subtract(2.8, 0.6)).to be_within(0.5).of(2.5)
+    end
+
+    it "subtracts a float and an integer" do
+      expect(calculator.subtract(1.2, 2)).to be_within(0.5).of(-1)
+    end
+  end
+
+  describe "#multiply" do
+    it "raises error if only one value is passed as argument" do
+      expect{ Calculator.new.multiply(1) }.to raise_error(ArgumentError)
+    end
+
+    it "raises error if more than two values are passed as argument" do
+      expect{ Calculator.new.multiply(1, 2, 3) }.to raise_error(ArgumentError)
+    end
+
+    it "multiplies two positive numbers" do
+      expect(calculator.multiply(3, 2)).to eq(6)
+    end
+
+    it "multiplies two negative numbers" do
+      expect(calculator.multiply(-2, -3)).to eq(6)
+    end
+
+    it "multiplies a positive number and a negative number" do
+      expect(calculator.multiply(5, -2)).to eq(-10)
+    end
+
+    it "multiplies two floats" do
+      expect(calculator.multiply(2.8, 0.6)).to be_within(0.5).of(2)
+    end
+
+    it "multiplies a float and an integer" do
+      expect(calculator.multiply(1.2, 2)).to be_within(0.5).of(2.5)
+    end
+  end
+
+  describe "#divide" do
+    it "raises error if only one value is passed as argument" do
+      expect{ Calculator.new.divide(1) }.to raise_error(ArgumentError)
+    end
+
+    it "raises error if more than two values are passed as argument" do
+      expect{ Calculator.new.divide(1, 2, 3) }.to raise_error(ArgumentError)
+    end
+
+    it "divides two positive numbers" do
+      expect(calculator.divide(6, 2)).to eq(3)
+    end
+
+    it "divides two negative numbers" do
+      expect(calculator.divide(-6, -3)).to eq(2)
+    end
+
+    it "divides a positive number and a negative number" do
+      expect(calculator.divide(10, -2)).to eq(-5)
+    end
+
+    it "divides two floats" do
+      expect(calculator.divide(2.8, 0.6)).to be_within(0.5).of(4.5)
+    end
+
+    it "divide a float and an integer" do
+      expect(calculator.divide(2.4, 2)).to be_within(0.5).of(1)
+    end
```

Comment:
`divide` should raise an ArgumentError if second argument is less than equal to zero